android/src/main/java/io/ably/flutter/plugin/types/PlatformClientOptions.java

```diff
@@ -0,0 +1,14 @@
+package io.ably.flutter.plugin.types;
+
+import io.ably.lib.types.ClientOptions;
+
+public class PlatformClientOptions extends ClientOptions {
```

Comment:
Akin to my comment on the Objective-C code earlier, this does not feel like the right place for inheritance. I would use composition here as well. e.g. something adopting:

```java
interface PlatformClientOptions {
    ClientOptions getClientOptions();
    boolean hasAuthCallback();
}
```

or similar.

Reply:
c9b9c89bd121150fe50d02d2fc03bd8216ad0fbe